### PR TITLE
feat(docs): Link to main sections from `/docs/`

### PR DESCRIPTION
**Pull request by fk**
TL;DR: Link the pages on https://www.gatsbyjs.org/docs/ marked in the screenshot below, add the required landing pages, and also link the corresponding items in the "Docs" sidebar navigation:

image

- add landing pages for
  - Guides
  - API Reference
  - Releases & Migration
  - Conceptual Guide
  - Advanced Tutorials
  to `docs/docs/`; their intro paragraph is taken from the copy at `www/src/pages/docs/index` / https://www.gatsbyjs.org/docs/
- link those in `www/src/data/doc-links.yaml` and `www/src/pages/docs/index`
  - `/docs/guides/`
  - `/docs/api-reference/`
  - `/docs/releases-and-migration/`
  - `/docs/conceptual-guide/`
  - `/docs/advanced-tutorials/`
- link „Ecosystem“ and „Advanced Tutorials“ in `www/src/pages/docs/index`

Slightly tweak `www/src/templates/template-docs-markdown`: Change „Guides in this section“ headline preceding the list of child articles to say „In this section“, thus making it work for non „Guides“ articles (e.g. the newly added landing pages):

image

image

/cc @calcsam @marcysutton

### CODE REVIEW

**Comment by fk** [comment]
Should have renamed `[[guidelist]]` to `[[sectionToc]]` or something. 🤦‍♂️ 
Let's do that when we move to MDX (#12514).
